indidriver.c

```diff
@@ -1774,6 +1774,7 @@ void IUSaveConfigBLOB(FILE *fp, const IBLOBVectorProperty *bvp)
 
         encblob        = malloc(4 * bp->bloblen / 3 + 4);
         l              = to64frombits(encblob, bp->blob, bp->bloblen);
+        assert(l <= (4 * bp->bloblen / 3 + 4));
```

Comment:
This assert (and the others you added) should part of the guarantees of `to64frombits()` and not a verification from the caller point of view. So move your `assert()` to that function.
Also, `encblob` could be returned as `null` by `malloc`, and that situation is not properly managed. Could you add a bailout path for this?

Reply:
I don't quite agree with that its to64frombits obligation to check memory size since it have no information about it. But your second point I will fix.

Reply:
I completely agree on `to64frombits` not having the size information. I think that's a shortcoming of its prototype, like `memcpy` and `strcpy` had in the past. Now, there is `dlen` inside `to64frombits`, which does a allocation estimation and could be use for that purpose.
I propose you add a new prototype, `to64frombits_s`, that takes the size of the output buffer as argument. There you may assert on the size of the `encblob` destination. We can then deprecate the use of `to64frombits` progressively afterwards. What do you think?

Reply:
The only rework in that case would be to call `to64frombits` from `to64frombits_s` after asserting the size of the destination using the same allocation length estimation done for `dlen`.

Reply:
Sounds like a good idea, ill have a look